### PR TITLE
`azurerm_monitor_aad_diagnostic_setting` - deprecate `enabled_log.retention_policy` block

### PR DESCRIPTION
**Pull request by teowa**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description
From API response:
```
=== RUN   TestAccMonitorAADDiagnosticSetting/basic/eventhubDefault
    testcase.go:180: Step 1/2 error: Error running apply: exit status 1
        Error: creating Diagnostic Setting (Diagnostic Setting Name: "acctest-DS-241029231212899469"): unexpected status 400 (400 Bad Request) with response: {"code":"BadRequest","message":"Diagnostic settings does not support retention for new diagnostic settings."}
          with azurerm_monitor_aad_diagnostic_setting.test,
          on terraform_plugin_test.tf line 52, in resource "azurerm_monitor_aad_diagnostic_setting" "test":
          52: resource "azurerm_monitor_aad_diagnostic_setting" "test" {
```

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

- env.ARM_FIVEPOINTZERO_BETA=false
image

- env.ARM_FIVEPOINTZERO_BETA=true
image

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_monitor_aad_diagnostic_setting` - deprecate `enabled_log.retention_policy` block

This is a (please select all that apply):

- [ ] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [x] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/monitor/monitor_aad_diagnostic_setting_resource_test.go [hunk 1]
```diff
@@ -128,6 +129,42 @@ func testAccMonitorAADDiagnosticSetting_updateToEnabledLog(t *testing.T) {
 	data := acceptance.BuildTestData(t, "azurerm_monitor_aad_diagnostic_setting", "test")
 	r := MonitorAADDiagnosticSettingResource{}
 
+	if !features.FivePointOhBeta() {
```
Comment (mbfrahry) [inline internal/services/monitor/monitor_aad_diagnostic_setting_resource_test.go:132]: New test but update config to have both

File: website/docs/r/monitor_aad_diagnostic_setting.html.markdown [hunk 2]
```diff
@@ -97,6 +97,8 @@ A `enabled_log` block supports the following:
 
 * `retention_policy` - (Required) A `retention_policy` block as defined below.
 
+-> **NOTE:** Azure does not support retention for new Azure Active Directory Diagnostic Settings. The `retention_policy` block is deprecated and will be removed in 5.0 version of the provider.
```
Comment (mbfrahry) [inline website/docs/r/monitor_aad_diagnostic_setting.html.markdown:100]: We can remove this as the deprecation message is sufficient

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @teowa, this PR looks good but there are a couple pieces that could be improved that I've called out

File: internal/services/monitor/monitor_aad_diagnostic_setting_resource_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (teowa) [reply internal/services/monitor/monitor_aad_diagnostic_setting_resource_test.go]: Thanks for reviewing this, I have changed to update to with/without retention in test config.  

> leave the existing configs untouched until we remove them in 5.0

The breaking change has happened at API side, we are unable to pass acctest if keep `retention_policy` enabled. So we must remove these `retention_policy`.

File: website/docs/r/monitor_aad_diagnostic_setting.html.markdown (hunk 2 above)
Comment (teowa) [reply website/docs/r/monitor_aad_diagnostic_setting.html.markdown]: updated

File: website/docs/r/monitor_aad_diagnostic_setting.html.markdown [hunk 3]
```diff
@@ -95,7 +95,7 @@ A `enabled_log` block supports the following:
 
 * `category` - (Required) The log category for the Azure Active Directory Diagnostic.
 
-* `retention_policy` - (Required) A `retention_policy` block as defined below.
+* `retention_policy` - (Optional) A `retention_policy` block as defined below.
```
Comment (mbfrahry) [inline website/docs/r/monitor_aad_diagnostic_setting.html.markdown:98]: The documentation for this field can actually be removed as we don't want anyone to use this going forward

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @teowa, I missed one more thing from our [breaking changes guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/guide-breaking-changes.md) and it's that this field should be removed from the documentation so people don't use a deprecated field going forward

File: website/docs/r/monitor_aad_diagnostic_setting.html.markdown (hunk 3 above)
Comment (teowa) [reply website/docs/r/monitor_aad_diagnostic_setting.html.markdown]: updated

**Pull request merged**